tools/xtce_generation/telemetry_generator.py

```diff
@@ -302,7 +284,20 @@ def create_parameters_from_dataframe(self, parameter_set, pkt):
 
         return parameter_set
 
-    def generate_telemetry_xml(self, output_xml_path):
+    def generate_telemetry_xml(
+        self, output_xml_path, mnemonic_names, science_packet_name="SciencePacket"
+    ):
+        """
+        Create and output an XTCE file based on the data within the class.
+        Parameters
+        ----------
+        output_xml_path: the path for the final xml file
+        mnemonic_names: a list or single name for mnemonics for extracting data types
```

Comment:
So this is where we give in names of science data that are specific to each instrument such as `SCIENCE_DATA` or `HIST_DATA` field in packet definition?

Reply:
Yes, exactly!